Datadog crash when attempting to create dashboards

Hi there,

We're running into an issue when trying to create dashboards with widgets in Datadog, the dashboards were initially created in the Datadog UI & since have been imported using Terraformer so we can manage them via terraform going forward.

Similar issue to [#1830](https://github.com/DataDog/terraform-provider-datadog/issues/1830) 

### Terraform Version

`1.4.2.`

### Affected Resource(s)

- datadog_dashboard

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files

Can't share the HCL in this forum, unfortunately, even if GPG encrypted.

### Debug Output

Can't share the full Debug output in this forum, unfortunately, even if GPG encrypted. Some of it is below with specifics removed

`│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).PlanResourceChange call. The plugin logs may contain more details.
╵
│ Error: Request cancelled
│ 
│ The plugin.(*GRPCProvider).PlanResourceChange request was cancelled.`

### Panic Output

`goroutine 82 [running]:
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogTemplateVariablePresets(0xc00237a978)
        github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:533 +0x6e5
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogDashboard(0xc002fcd800)
        github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:375 +0x665
github.com/terraform-providers/terraform-provider-datadog/datadog.resourceDatadogDashboardCreate({0x23a04a8, 0xc002fa9140}, 0xc002fcd800, {0x1e1f420?, 0xc0006ca990})
        github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:148 +0xbe
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc000294d20, {0x23a04e0, 0xc0001a6c00}, 0xd?, {0x1e1f420, 0xc0006ca990})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:707 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000294d20, {0x23a04e0, 0xc0001a6c00}, 0xc000ab36c0, 0xc000886c00, {0x1e1f420, 0xc0006ca990})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:837 +0xa7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000400000, {0x23a04e0?, 0xc0001a6ab0?}, 0xc00188e4b0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:1021 +0xe3c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0003e2f00, {0x23a04e0?, 0xc0001a60c0?}, 0xc0020f4070)
        github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x20c11e0?, 0xc0003e2f00}, {0x23a04e0, 0xc0001a60c0}, 0xc0020f4000, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002701e0, {0x23a32e0, 0xc000596340}, 0xc001614ea0, 0xc000380600, 0x2e93780, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1340 +0xd13
google.golang.org/grpc.(*Server).handleStream(0xc0002701e0, {0x23a32e0, 0xc000596340}, 0xc001614ea0, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1713 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.50.1/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.50.1/server.go:963 +0x28a
panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 27 [running]:
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogTemplateVariablePresets(0xc000b8a978)
        github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:533 +0x6e5
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogDashboard(0xc0015e6680)
        github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:375 +0x665
github.com/terraform-providers/terraform-provider-datadog/datadog.resourceDatadogDashboardCreate({0x23a04a8, 0xc002159740}, 0xc0015e6680, {0x1e1f420?, 0xc000abfef0})
        github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:148 +0xbe
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc000308d20, {0x23a04e0, 0xc00036d020}, 0xd?, {0x1e1f420, 0xc000abfef0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:707 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000308d20, {0x23a04e0, 0xc00036d020}, 0xc000847450, 0xc000a76580, {0x1e1f420, 0xc000abfef0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:837 +0xa7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000284018, {0x23a04e0?, 0xc00036cde0?}, 0xc000a36370)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:1021 +0xe3c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0003ecf00, {0x23a04e0?, 0xc00036c450?}, 0xc0003cd650)
        github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x20c11e0?, 0xc0003ecf00}, {0x23a04e0, 0xc00036c450}, 0xc0003cd5e0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00042e000, {0x23a32e0, 0xc0004096c0}, 0xc00215a6c0, 0xc000718600, 0x2e93780, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1340 +0xd13
google.golang.org/grpc.(*Server).handleStream(0xc00042e000, {0x23a32e0, 0xc0004096c0}, 0xc00215a6c0, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1713 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.50.1/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.50.1/server.go:963 +0x28a

Error: The terraform-provider-datadog_v3.23.0 plugin crashed!`

### Expected Behavior

Datadog dashboards should have been created with widgets attached to each.

### Actual Behavior

One Datadog dashboard has been created but none of the others.

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Some of the dashboards have many widgets, these were created manually within the Datadog UI but using terraformer we have imported the configuration files so we can manage & deploy via terraform in the future.

### References

Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:

-    [#1830](https://github.com/DataDog/terraform-provider-datadog/issues/1830)

--- a/datadog/resource_datadog_dashboard.go
+++ b/datadog/resource_datadog_dashboard.go
@@ -420,6 +420,9 @@ func getTemplateVariableSchema() map[string]*schema.Schema {
 func buildDatadogTemplateVariables(terraformTemplateVariables *[]interface{}) *[]datadogV1.DashboardTemplateVariable {
 	datadogTemplateVariables := make([]datadogV1.DashboardTemplateVariable, len(*terraformTemplateVariables))
 	for i, ttv := range *terraformTemplateVariables {
+		if ttv == nil {
+			continue
+		}
 		terraformTemplateVariable := ttv.(map[string]interface{})
 		var datadogTemplateVariable datadogV1.DashboardTemplateVariable
 		if v, ok := terraformTemplateVariable["name"].(string); ok && len(v) != 0 {
@@ -544,6 +547,9 @@ func buildDatadogTemplateVariablePresets(terraformTemplateVariablePresets *[]int
 			datadogTemplateVariablePresetValues := make([]datadogV1.DashboardTemplateVariablePresetValue, len(templateVariablePresetValues))
 
 			for j, tvp := range templateVariablePresetValues {
+				if tvp == nil {
+					continue
+				}
 				templateVariablePresetValue := tvp.(map[string]interface{})
 				var datadogTemplateVariablePresetValue datadogV1.DashboardTemplateVariablePresetValue
 

--- a/datadog/resource_datadog_dashboard.go
+++ b/datadog/resource_datadog_dashboard.go
@@ -530,6 +530,9 @@ func buildDatadogTemplateVariablePresets(terraformTemplateVariablePresets *[]int
 	datadogTemplateVariablePresets := make([]datadogV1.DashboardTemplateVariablePreset, len(*terraformTemplateVariablePresets))
 
 	for i, tvp := range *terraformTemplateVariablePresets {
+		if tvp == nil {
+			continue
+		}
 		templateVariablePreset := tvp.(map[string]interface{})
 		var datadogTemplateVariablePreset datadogV1.DashboardTemplateVariablePreset
 

